File: components/core/src/main/java/org/storm/core/input/action/ReadonlyActionManager.java
package org.storm.core.input.action;

import org.storm.core.exception.ActionManagerException;

import java.util.Map;

/**
 * An implementation of an ActionManager which only allows for checking the state of an action.
 */
public class ReadonlyActionManager extends ActionManager {

  public ReadonlyActionManager(Map<String, Boolean> state) {
    super(state);
  }

  @Override
  public void add(String action) {
    throw new ActionManagerException("add not implemented for %s", this.getClass().getName());
  }

  @Override
  public void remove(String action) {
    throw new ActionManagerException("remove not implemented for %s", this.getClass().getName());
  }

  @Override
  public void startUsing(String action) {
    throw new ActionManagerException("startUsing not implemented for %s", this.getClass().getName());
  }

  @Override
  public void stopUsing(String action) {
    throw new ActionManagerException("stopUsing not implemented for %s", this.getClass().getName());
  }

}

File: components/core/src/main/kotlin/org/storm/core/input/action/ReadonlyActionManager.kt
package org.storm.core.input.action

import org.storm.core.exception.ActionManagerException

/**
 * An implementation of an ActionManager which only allows for checking the state of an action.
 */
class ReadonlyActionManager(state: MutableMap<String, Boolean>) : ActionManager(state) {
  override fun add(action: String) {
    throw ActionManagerException("add not implemented for ${this.javaClass.name}")
  }

  override fun remove(action: String) {
    throw ActionManagerException("remove not implemented for ${this.javaClass.name}")
  }

  override fun startUsing(action: String) {
    throw ActionManagerException("startUsing not implemented for ${this.javaClass.name}")
  }

  override fun stopUsing(action: String) {
    throw ActionManagerException("stopUsing not implemented for ${this.javaClass.name}")
  }
}
